src/com/t_oster/liblasercut/drivers/GenericGcodeDriver.java

```diff
@@ -785,6 +821,17 @@ else if (this.port != null)
 
   }
 
+
+  /* sendJobPrepare() and sendJobFinish() can be overrided in children to
+   * perform device-specific setup before and after sending job over serial
+   * line
+   */
+  protected void sendJobPrepare() throws IOException {
+  }
+
+  protected void sendJobFinish() throws IOException {
+  }
+
```

Comment:
There is already `writeInitialiuationCode` and `writeShutdownCode` for exactly this purpose. Why another set of methods? You can override them and if you need you still can call `super.writeInitializationCode()` to get the behavior of the parent driver.